### DISCUSSION TITLE
Make Contribution User Story Better

### DISCUSSION THREAD

**Discussion by davidbuckleyni** [dotnet/maui#13927] [General]
### Description

As a developer, I would like to be able to build MAUI and compile MAUI locally on a windows machine at present when following the directions here it is out of date and the solution does not compile when check out from the main branch.

https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md

### Steps to Reproduce

1. Complete the steps in the development guide first.
2. Clone the repo from the main branch here https://github.com/dotnet/maui.git
3. Make sure all the sdks are installed
4. Load the solution file Microsoft.Maui
5. Clean and do a retore compile and build
6.Build fails with over 13819 errors

![image](https://user-images.githubusercontent.com/5619229/170894344-53fada93-85df-4642-9c94-cd3002811176.png)

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

Maui Main Branch Build

### Did you find any workaround?

None

**Comment by drasticactions** [comment]
Did you follow the directions?

I don't disagree that those directions should be updated (It still references RC 2 and Preview 14) but for Main it should work, it does for me.

Did you run the Cake script? The cake script launches VS when it's done building. If you just launch it without that, it (most likely) won't work because of the amount of projects and such.

**Comment by davidbuckleyni** [comment]
Yeah i did but i was using main branch maybe that was the issue @drasticactions ill try again tommrrow with rc2. 

Which is the ga branch though?

**Comment by davidbuckleyni** [comment]
Ok I did what you said ran the CMake script it seem to work but then I got this warning.
@drasticactions 
![image](https://user-images.githubusercontent.com/5619229/170983710-43469da1-eef4-4848-b598-2b9b564ea42e.png)

**Comment by davidbuckleyni** [comment]
Also to add how on earth is this productive for the development team if they have to run CMake every time do they have a special version of visual studio that means they don't need to I can see your dev teams getting frustrated having to run the CMake every change and how does one see their changes @drasticactions.

**Comment by Eilon** [comment]
BTW this doesn't use CMake, it uses Cake.

**Comment by davidbuckleyni** [comment]
> BTW this doesn't use CMake, it uses Cake.

Yeah I know that I don't mind that but it must be very hard for even internal dev teams to have a happy path if it makes it that hard for us to contribute.
